core/ecschema-metadata/src/SchemaFormatsProvider.ts

```diff
@@ -0,0 +1,122 @@
+import { UnitSystemKey } from "@itwin/core-quantity";
+
+/**
+ * Provides default formats and kind of quantities from a given SchemaContext or SchemaLocater.
+ * @beta
+ */
+export class SchemaFormatsProvider implements FormatsProvider {
+  private _context: SchemaContext;
+  private _unitSystem: UnitSystemKey;
+  public onFormatsChanged = new BeUiEvent<string[]>();
+  /**
+   *
+   * @param contextOrLocater The SchemaContext or a different ISchemaLocater implementation used to retrieve the schema. The SchemaContext
+   * class implements the ISchemaLocater interface. If the provided locater is not a SchemaContext instance a new SchemaContext will be
+   * created and the locater will be added.
+   * @param _unitExtraData Additional data like alternate display label not found in Units Schema to match with Units; Defaults to empty array.
+   */
+  constructor(contextOrLocater: ISchemaLocater, unitSystem: UnitSystemKey) {
+    if (contextOrLocater instanceof SchemaContext) {
+      this._context = contextOrLocater;
+    } else {
+      this._context = new SchemaContext();
+      this._context.addLocater(contextOrLocater);
+    }
+    this._unitSystem = unitSystem;
+  }
+
+  public get context() { return this._context; }
+  public get unitSystem() { return this._unitSystem; }
+
+  public set unitSystem(unitSystem: UnitSystemKey) {
+    this._unitSystem = unitSystem;
+    this.onFormatsChanged.raiseEvent([this._unitSystem]);
+  }
+
+  private async getKindOfQuantityFormatFromSchema(itemKey: SchemaItemKey, unitSystem: UnitSystemKey): Promise<SchemaItemFormatProps | undefined> {
+    const kindOfQuantity = await this._context.getSchemaItem(itemKey, KindOfQuantity);
+
+    if (!kindOfQuantity) {
+      return undefined;
+    }
+    // Find the first presentation format that matches the provided unit system. Else, use the first entry.
+    if (unitSystem) {
+      const unitSystemGroupNames = getUnitSystemGroupNames(unitSystem);
+      const presentationFormats = kindOfQuantity.presentationFormats;
+      for (const system of unitSystemGroupNames) {
+        for (const format of presentationFormats) {
+          const unit = format.units && format.units[0][0];
+          if (!unit) {
+            continue;
+          }
+          const currentUnitSystem = await unit.unitSystem;
+          if (currentUnitSystem && currentUnitSystem.name.toUpperCase() === system) {
+            return getFormatProps(format);
+          }
+        }
+      }
+      return undefined;
+    }
+    const defaultFormat = kindOfQuantity.defaultPresentationFormat;
+    if (!defaultFormat) {
+      return undefined;
+    }
+    return getFormatProps(defaultFormat);
+  }
+
+
+  /**
+   * Retrieves a Format from a SchemaContext. If the format is part of a KindOfQuantity, the first presentation format in the KindOfQuantity that matches the current unit system will be retrieved.
+   * @param name The full name of the Format or KindOfQuantity.
```

Comment:
I think docs of this argument should be added to the `FormatsProvider` interface.

Reply:
added